app/controllers/benefits_applications_controller.rb

```diff
@@ -79,12 +79,29 @@ def create_member
     save_member(benefit_app, @member_form)
   end
 
+  def sign_benefits_app
+    @benefit_app_form = current_benefit_app
+    render :sign_benefits_app
+  end
+
+  def save_signature
+    @benefit_app_form = current_benefit_app
+    current_date = Time.now
+    if @benefit_app_form.update(benefits_permitted_params.merge({signed_date: current_date}))
+      redirect_to root_path
+    end
+  end
+
   def validate_application
     benefit_app = current_benefit_app
 
     if benefit_app&.primary_member.present?
       benefit_app.update(submitted_at: Date.today)
-      redirect_to root_url
+      if benefit_app.signature?
```

Comment:
We could make this into a method that checks both `signature` and `signed_date`, something like:

```diff
diff --git a/app/controllers/benefits_applications_controller.rb b/app/controllers/benefits_applications_controller.rb
index 13d231d..f9bb7b9 100644
--- a/app/controllers/benefits_applications_controller.rb
+++ b/app/controllers/benefits_applications_controller.rb
@@ -97,7 +97,8 @@ class BenefitsApplicationsController < ApplicationController

     if benefit_app&.primary_member.present?
       benefit_app.update(submitted_at: Date.today)
-      if benefit_app.signature?
+
+      if benefit_app.was_signed?
         redirect_to root_path
       else
         redirect_to sign_benefits_app_path
diff --git a/app/models/benefit_app.rb b/app/models/benefit_app.rb
index 3d799e2..1cb2c58 100644
--- a/app/models/benefit_app.rb
+++ b/app/models/benefit_app.rb
@@ -17,4 +17,8 @@ class BenefitApp < ApplicationRecord
       "No Primary Member"
     end
   end
+
+  def was_signed?
+    signature.present? and signed_date.present?
+  end
 end
\ No newline at end of file
diff --git a/spec/factories/benefit_app.rb b/spec/factories/benefit_app.rb
index b8c4f51..6c97dc0 100644
--- a/spec/factories/benefit_app.rb
+++ b/spec/factories/benefit_app.rb
@@ -29,5 +29,11 @@ FactoryBot.define do
     factory :benefit_app_with_non_primary_member do
       primary_member { build(:member, is_primary: false)}
     end
+
+    factory :signed_benefit_app do
+      primary_member { create(:primary_member) }
+      signature { primary_member.full_name }
+      signed_date { DateTime.now }
+    end
   end
 end
\ No newline at end of file
diff --git a/spec/models/benefit_app_spec.rb b/spec/models/benefit_app_spec.rb
index 0067c19..e21276b 100644
--- a/spec/models/benefit_app_spec.rb
+++ b/spec/models/benefit_app_spec.rb
@@ -38,6 +38,19 @@ RSpec.describe BenefitApp, type: :model do
       expect(benefit_app_without_address).not_to be_valid
     end

+    describe "#was_signed" do
+      subject(:unsigned_benefit_app) { create(:benefit_app) }
+      subject(:signed_benefit_app) { create(:signed_benefit_app)}
+
+      it "checks signed apps" do
+        expect(signed_benefit_app.was_signed?).to be_truthy
+      end
+
+      it "checks unsigned apps" do
+        expect(unsigned_benefit_app.was_signed?).to be_falsey
+      end
+    end
+
     context "with associating members" do
       it "expects its primary member to be marked as one" do
         expect(benefit_app_with_primary_member.primary_member.is_primary).to be_truthy

```

Reply:
Heh, just noticed my use of `DateTime.now` instead of `Date.today`.